src/pyJiraCli/jira_server.py

```diff
@@ -303,42 +293,99 @@ def _login_with_password(self, user:str, pw:str) -> Ret:
 
             self._jira_obj.verify_ssl = False
 
-        except (exceptions.JIRAError, reqex.ConnectionError, urlex.MaxRetryError) as e:
+        except (exceptions.JIRAError,
+                urlex.MaxRetryError,
+                reqex.ConnectionError,
+                reqex.MissingSchema,
+                reqex.InvalidSchema,
+                reqex.InvalidURL) as e:
             #print error
+            ret_status = Ret.CODE.RET_ERROR_JIRA_LOGIN
+
             if isinstance(e, exceptions.JIRAError):
                 print(e.text)
-            else:
-                print(str(e))
 
-            ret_status = Ret.CODE.RET_ERROR_JIRA_LOGIN
+            elif isinstance(e, (reqex.InvalidSchema,
+                                reqex.MissingSchema,
+                                reqex.InvalidURL)):
+                ret_status = Ret.CODE.RET_ERROR_INVALID_URL
 
-        if user is None:
-            ret_status = Ret.CODE.RET_ERROR_JIRA_LOGIN
+            else:
+                print(str(e))
 
         self._user = user
 
         return ret_status
 
-    def _get_server_url(self) -> Tuple[str, None]:
-        """ Get the server url from the encrypted files.
+################################################################################
+# Functions
+################################################################################
+def _get_user_input() -> Tuple[str, str]:
+    """Prompt the user to enter a username and a password.
+    The password input is masked with '*' characters.
+
+    Returns:
+        Tuple[str, str]: A tuple containing the username and the password.
+    """
+    username = input("Enter your username: ")
+
+    print("Enter your password: ", end="", flush=True)
+    password = _get_password()
+
+    return username, password
+
+if os.name == 'nt':
+    def _get_password() -> str:
```

Comment:
I believe this functionality already exists in Python. Take a look at `getpass`

Reply:
The getpass function does not mask the input with '*' but instead displays nothing when a password is entered. To give the user a sense of what was entered, this approach was used to at least display the lentgh of the entered password with '*'